openqaoa/problems/problem.py

```diff
@@ -118,6 +140,28 @@ def __init__(self, n, terms, weights, clean_terms_and_weights=False):
         self.constant = constant
         self.n = n
 
+        # attribute to store the problem instance, it will be checked if it is json serializable in the __setattr__ method
+        self.problem_instance = problem_instance
+
+        # Initialize the metadata dictionary 
+        self.metadata = {}
+
+    def __iter__(self):
+        for key, value in self.__dict__.items():
+            # remove "_" from the beginning of the key if it exists
+            yield (key[1:] if key.startswith("_") else key, value)
+
+    def __setattr__(self, __name, __value):
+        # check if problem_instance is json serializable, also check if metadata is json serializable
+        if __name == "problem_instance" or __name == "metadata":
+            try:
+                _ = json.dumps(__value)
+            except Exception as e:
+                raise e    
```

Comment:
I think it is easier to check if the metadata is serializable by doing that, @Q-lds what do you think?

Reply:
Sure, good idea!